File: backend/OrganizedLife.API/Models/Entities/Category.cs
namespace OrganizedLife.API.Models.Entities;

public class Category
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Icon { get; set; } = string.Empty;
    public string Color { get; set; } = "#6B7280";
    public TransactionType Type { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    // Foreign key (null for system-wide categories)
    public int? UserId { get; set; }

    // Navigation properties
    public User? User { get; set; }
    public ICollection<Transaction> Transactions { get; set; } = new List<Transaction>();
}

File: frontend/src/types/category.ts
export enum CategoryType {
  INCOME = 'income',
  EXPENSE = 'expense',
}

export interface Category {
  id: number
  name: string
  color: string
  icon: string
  type: CategoryType
  userId: number
}

export interface CreateCategoryDTO {
  name: string
  color: string
  icon: string
  type: CategoryType
}

export interface UpdateCategoryDTO extends CreateCategoryDTO {
  id: number
}
